ui/core-react/src/test/tabs/Tabs.test.tsx

```diff
@@ -2,230 +2,222 @@
+
+    await theUserTo.keyboard("{enter}");
+    expect(classesFromElement(screen.getByRole("tab", {name: "label 1"}))).to.include("core-active");
     spyActivate.should.have.been.calledOnceWithExactly(0);
-    wrapper.unmount();
   });
 
-  it("Space key in activates tab", () => {
+  it("Space key in activates tab", async () => {
     const spyActivate = sinon.spy();
-    const wrapper = mount<Tabs>(<Tabs orientation={Orientation.Vertical} mainClassName="" labels={["label 1", "label 2", "label 3"]} activeIndex={1} onActivateTab={spyActivate} />);
-    expect(wrapper.state().activeIndex).to.eq(1);
-    const label = wrapper.find("a").at(2);
-    label.simulate("keydown", { key: " " });
-    label.simulate("keyup", { key: " " });
-    wrapper.update();
-    expect(wrapper.state().activeIndex).to.eq(2);
-    spyActivate.should.have.been.calledOnceWithExactly(2);
-    wrapper.unmount();
+    render(<VerticalTabs labels={["label 1", "label 2", "label 3"]} activeIndex={1} onActivateTab={spyActivate} />);
+    const label = screen.getByRole("button", {name: "label 2"});
+    await theUserTo.type(label, "{arrowup}");
+    spyActivate.resetHistory();
+
+    await theUserTo.keyboard(" ");
+    expect(classesFromElement(screen.getByRole("tab", {name: "label 1"}))).to.include("core-active");
+    spyActivate.should.have.been.calledOnceWithExactly(0);
   });
 
   ///
 
-  it("Supports updating labels & orientation", () => {
+  it("Supports updating labels & orientation", async () => {
     const { container, getAllByRole, rerender } = render(<Tabs orientation={Orientation.Vertical} mainClassName="" labels={["label 1", "label 2", "label 3"]} activeIndex={1} />);
     container.focus();
     let tabButtons = getAllByRole("button");
     expect(tabButtons.length).to.eq(3);
     let label = screen.getByText("label 2");
     // verify they're vertical by using arrow up to change focus
-    userEvent.type(label, "{arrowup}");
+    await theUserTo.type(label, "{arrowup}");
     expect(document.activeElement).to.eq(tabButtons[0]);
 
     rerender(<Tabs orientation={Orientation.Horizontal} mainClassName="" labels={["label 1", "label 2", "label 3", "label 4"]} activeIndex={1} />);
     tabButtons = getAllByRole("button");
     expect(tabButtons.length).to.eq(4);
     label = screen.getByText("label 2");
-    userEvent.type(label, "{enter}"); // focus in the tab
-    userEvent.type(label, "{arrowup}");
+    await theUserTo.type(label, "{enter}"); // focus in the tab
+    await theUserTo.type(label, "{arrowup}");
     // arrow up does not change focus because they're horizontal
     expect(document.activeElement).to.eq(tabButtons[1]);
   });
 
   it("Supports updating activeIndex", async () => {
-    const { container, getByText, getAllByRole, rerender } = render(<Tabs orientation={Orientation.Vertical} mainClassName="" labels={["label 1", "label 2", "label 3"]} activeIndex={0} />);
-    const tabsInstance = findInstance(container.firstChild);
```

Comment:
Testing the user visible output instead of the internal state, so I completely removed "ReactInstance".